frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -558,52 +562,69 @@ const UpdatesType: React.FC<UpdatesTypeProps> = ({ children }) => {
   const updatedMCPNodes =
-    MCPUpdatingTime > updateStartedTime ? machineConfigPool?.status?.updatedMachineCount : 0;
+    updatedMachineCountReady || MCPUpdatingTime > updateStartedTime
+      ? machineConfigPool?.status?.updatedMachineCount
+      : 0;
   const percentMCPNodes = calculatePercentage(updatedMCPNodes, totalMCPNodes);
   const { t } = useTranslation();
-  return hideIfComplete && percentMCPNodes === 100 ? null : (
-    <UpdatesGroup divided={divided}>
-      <UpdatesType>
-        <Link to={`/k8s/cluster/nodes?rowFilter-node-role=${machineConfigPool.metadata.name}`}>
-          {t('public~{{name}} Nodes', { name })}
-        </Link>
-        {name !== 'Master' && (
-          <FieldLevelHelp>
-            {t(
-              'public~{{name}} {{resource}} may continue to update after the update of Master {{resource}} and {{resource2}} are complete.',
-              {
-                name,
-                resource: NodeModel.labelPlural,
-                resource2: ClusterOperatorModel.labelPlural,
-              },
+  return hideIfComplete && percentMCPNodes === 100
+    ? null
+    : machineConfigOperatorLoaded && (
+        <UpdatesGroup divided={divided}>
+          <UpdatesType>
+            <Link to={`/k8s/cluster/nodes?rowFilter-node-role=${machineConfigPool.metadata.name}`}>
+              {t('public~{{name}} Nodes', { name })}
+            </Link>
+            {name !== 'Master' && (
+              <FieldLevelHelp>
+                {t(
+                  'public~{{name}} {{resource}} may continue to update after the update of Master {{resource}} and {{resource2}} are complete.',
+                  {
+                    name,
+                    resource: NodeModel.labelPlural,
+                    resource2: ClusterOperatorModel.labelPlural,
+                  },
+                )}
+              </FieldLevelHelp>
             )}
-          </FieldLevelHelp>
-        )}
-      </UpdatesType>
-      <UpdatesBar>
-        <Progress
-          title={t('public~{{updatedMCPNodes}} of {{totalMCPNodes}}', {
-            updatedMCPNodes,
-            totalMCPNodes,
-          })}
-          value={!_.isNaN(percentMCPNodes) ? percentMCPNodes : null}
-          size={ProgressSize.sm}
-          variant={percentMCPNodes === 100 ? ProgressVariant.success : null}
-        />
-      </UpdatesBar>
-    </UpdatesGroup>
-  );
+          </UpdatesType>
+          <UpdatesBar>
+            <Progress
+              title={t('public~{{updatedMCPNodes}} of {{totalMCPNodes}}', {
+                updatedMCPNodes,
+                totalMCPNodes,
+              })}
+              value={!_.isNaN(percentMCPNodes) ? percentMCPNodes : null}
+              size={ProgressSize.sm}
+              variant={percentMCPNodes === 100 ? ProgressVariant.success : null}
+            />
+          </UpdatesBar>
+        </UpdatesGroup>
+      );
 };
 
 const OtherNodes: React.FC<OtherNodesProps> = ({
+  desiredVersion,
```

Comment:
Consider React Context to avoid needing to pass so many props down through the component tree. (Not required for this fix, just an observation.)

https://reactjs.org/docs/context.html